src/main/java/com/spotify/dns/AbstractChangeNotifier.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2012-2015 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.dns;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A helper for implementing the {@link ChangeNotifier} interface.
+ */
+abstract class AbstractChangeNotifier<T> implements ChangeNotifier<T> {
```

Comment:
petter: yes, probably a bit overkill if we only want a single listener. shouldn't multiple listeners be able to listen on the same dns updates though? but perhaps that's not a common usecase.